torchtext/datasets/ag_news.py

```diff
@@ -30,11 +31,18 @@
 
 @_add_docstring_header(num_lines=NUM_LINES, num_classes=4)
 @_create_dataset_directory(dataset_name=DATASET_NAME)
-@_wrap_split_argument(('train', 'test'))
-def AG_NEWS(root, split):
-    path = download_from_url(URL[split], root=root,
-                             path=os.path.join(root, split + ".csv"),
-                             hash_value=MD5[split],
-                             hash_type='md5')
-    return _RawTextIterableDataset(DATASET_NAME, NUM_LINES[split],
-                                   _create_data_from_csv(path))
+@_wrap_split_argument(("train", "test"))
+def AG_NEWS(root: str, split: Union[Tuple[str], str]):
```

Comment:
NIT: will the type for `split` ever be `Tuple[str]`? Because as far as I understand the `@_wrap_split_argument` decorator will make `n` number of calls to this method where `n` is the number of splits in the `split` argument.

cc @parmeet

Reply:
You could ask for the train and test splits for instance, right?

Reply:
Yes, the split can be either tuple or just string. A user can pass something like `(train,dev,test)` which returns all the splits or just 'train' which would return only train split. The decorator is precisely taking care of this i.e to handle both the scenarios.